### PR TITLE
NEXTEUROPA-6900 : Use separate form_alters wherever it is possible

### PR DESCRIPTION
**Pull request by senseca**

### CODE REVIEW

File: profiles/common/modules/features/communities/communities.module [hunk 1]
```diff
@@ -505,18 +505,18 @@ function communities_node_view($node, $view_mode, $langcode) {
...
   if (isset($form_state['build_info']['base_form_id']) && $form_state['build_info']['base_form_id'] == 'node_form') {
-    // If it's a node edit form.
     $context = og_context();
+    // We are not in a og context.
     if (!isset($context['gid'])) {
-      // We are not in a og context.
       if (isset($form[OG_CONTENT_ACCESS_FIELD])) {
-        $form['field_highlighted']['#access'] = FALSE;
         // Hide highlighted field.
+        $form['field_highlighted']['#access'] = FALSE;
```
Comment (pfrenssen) [inline profiles/common/modules/features/communities/communities.module]: Nice catch, makes sense to have the documentation above the functionality.

File: profiles/common/modules/features/multisite_workbench_moderation_view/multisite_workbench_moderation_view.module [hunk 2]
```diff
@@ -38,15 +38,12 @@ function multisite_workbench_moderation_view_views_pre_view(&$views) {
 }
 
 /**
- * Implements hook_form_alter().
+ * Implements hook_form_FOMR_ID_alter().
```
Comment (pfrenssen) [inline profiles/common/modules/features/multisite_workbench_moderation_view/multisite_workbench_moderation_view.module]: Typo: `FOMR`

File: profiles/common/modules/features/multisite_workbench_moderation_view/multisite_workbench_moderation_view.module [hunk 3]
```diff
@@ -38,15 +38,12 @@ function multisite_workbench_moderation_view_views_pre_view(&$views) {
...
-    if (!empty($form['force_transition']) && isset($form['force_transition'])) {
-
-      // Beware is dangerous to unset variables, choose better to hide them.
-      $form['force_transition']['#type'] = 'hidden';
-    }
+function multisite_workbench_moderation_view_form_views_form_workbench_moderation_moderate_all_page_alter(&$form, &$form_state, $form_id) {
+  if (!empty($form['force_transition']) && isset($form['force_transition'])) {
+    // Beware is dangerous to unset variables, choose better to hide them.
+    $form['force_transition']['#type'] = 'hidden';
   }
```
Comment (pfrenssen) [inline profiles/common/modules/features/multisite_workbench_moderation_view/multisite_workbench_moderation_view.module]: Wow this must be a finalist in the competition of "Longest function name ever"!

`multisite_workbench_moderation_view_form_views_form_workbench_moderation_moderate_all_page_alter()`

File: profiles/common/modules/features/news/news_og/news_og.module [hunk 4]
```diff
@@ -28,13 +28,9 @@ function news_og_menu() {
...
-function news_og_form_alter(&$form, $form_state, $form_id) {
-  switch ($form_id) {
-    case 'news_node_form':
-      $desc = $form['field_top_news']['und']['#description'];
-      $form['field_top_news']['und']['#description'] = $desc . " " . t('(The visibility of the news must be set to public to appear on the site homepage)');
-      break;
-  }
+function news_og_form_news_node_form_alter(&$form, $form_state, $form_id) {
+  $desc = $form['field_top_news']['und']['#description'];
+  $form['field_top_news']['und']['#description'] = $desc . " " . t('(The visibility of the news must be set to public to appear on the site homepage)');
```
Comment (pfrenssen) [inline profiles/common/modules/features/news/news_og/news_og.module]: There is a small bug here: the `$form_state` should be by reference. This was already there before, and currently this `$form_state` variable is unused so it doesn't affect the code. It's still a good idea to fix it, otherwise when people work on this code in the future and change the form state it will be very confusing why it doesn't work.

You can also remove the `$form_id` when converting a `hook_form_alter()` to a `hook_form_FORM_ID_alter()`. It's no longer needed then, even though it is still passed to the function.

File: profiles/common/modules/features/solr_config/solr_config.module [hunk 5]
```diff
@@ -42,26 +42,14 @@ function solr_config_apachesolr_exclude($entity_id, $entity_type, $row, $env_id)
...
-
-/**
- * Implements hook_apachesolr_query_alter().
- */
-function solr_config_apachesolr_query_alter($query) {
-  global $language;
-  // By default set the facet language to current language and neutral language.
-  $query->addFilter('ss_language', "(und OR " . $language->language . ")");
-}
-
```
Comment (pfrenssen) [inline profiles/common/modules/features/solr_config/solr_config.module]: Why was this removed?

File: profiles/common/modules/features/multisite_workbench_moderation_view/multisite_workbench_moderation_view.module (hunk 3 above)
Comment (senseca) [inline profiles/common/modules/features/multisite_workbench_moderation_view/multisite_workbench_moderation_view.module]: Lol! Exact!

File: profiles/common/modules/features/solr_config/solr_config.module (hunk 5 above)
Comment (senseca) [inline profiles/common/modules/features/solr_config/solr_config.module]: I don't know why either!
